fractal_tasks_core/tasks/calculate_2D_registration_image_based.py

```diff
@@ -0,0 +1,338 @@
+
+    # Read some parameters from metadata
+    coarsening_xy = metadata["coarsening_xy"]
+
+    # Get channel_index via wavelength_id.
+    # Intially only allow registration of the same wavelength
+    channel_ref: OmeroChannel = get_channel_from_image_zarr(
+        image_zarr_path=str(zarr_img_ref_cycle),
+        wavelength_id=wavelength_id,
+    )
+    channel_index_ref = channel_ref.index
+
+    channel_align: OmeroChannel = get_channel_from_image_zarr(
+        image_zarr_path=str(zarr_img_cycle_x),
+        wavelength_id=wavelength_id,
+    )
+    channel_index_align = channel_align.index
+
+    # Lazily load zarr array
+    data_reference_zyx = da.from_zarr(f"{zarr_img_ref_cycle}/{level}")[
+        channel_index_ref
+    ]
+    data_alignment_zyx = da.from_zarr(f"{zarr_img_cycle_x}/{level}")[
+        channel_index_align
+    ]
+
+    # Read FOV ROIs
+    FOV_ROI_table_ref = ad.read_zarr(
+        f"{zarr_img_ref_cycle}/tables/{roi_table}"
+    )
+    FOV_ROI_table_x = ad.read_zarr(f"{zarr_img_ref_cycle}/tables/{roi_table}")
+    logger.info(
+        f"Found {len(FOV_ROI_table_x)} ROIs in {roi_table=} to be processed."
+    )
+
+    # For each cycle, get the relevant info
+    # TODO: Add additional checks on ROIs?
+    if (FOV_ROI_table_ref.obs.index != FOV_ROI_table_x.obs.index).all():
+        raise ValueError(
+            "Registration is only implemented for ROIs that match between the "
+            "cycles (e.g. well, FOV ROIs). Here, the ROIs in the reference "
+            "cycles were {FOV_ROI_table_ref.obs.index}, but the ROIs in the "
+            "alignment cycle were {FOV_ROI_table_x.obs.index}"
+        )
+    # TODO: Make this less restrictive? i.e. could we also run it if different
+    # cycles have different FOVs? But then how do we know which FOVs to match?
+    # If we relax this, downstream assumptions on matching based on order
+    # in the list will break.
+
+    # Read pixel sizes from zattrs file for full_res
+    pxl_sizes_zyx = extract_zyx_pixel_sizes(
+        f"{zarr_img_ref_cycle}/.zattrs", level=0
+    )
+    pxl_sizes_zyx_cycle_x = extract_zyx_pixel_sizes(
+        f"{zarr_img_cycle_x}/.zattrs", level=0
+    )
+
+    if pxl_sizes_zyx != pxl_sizes_zyx_cycle_x:
+        raise ValueError(
+            "Pixel sizes need to be equal between cycles for registration"
+        )
+
+    # Create list of indices for 3D FOVs spanning the entire Z direction
+    list_indices_ref = convert_ROI_table_to_indices(
+        FOV_ROI_table_ref,
+        level=level,
+        coarsening_xy=coarsening_xy,
+        full_res_pxl_sizes_zyx=pxl_sizes_zyx,
+    )
+
+    list_indices_cycle_x = convert_ROI_table_to_indices(
+        FOV_ROI_table_x,
+        level=level,
+        coarsening_xy=coarsening_xy,
+        full_res_pxl_sizes_zyx=pxl_sizes_zyx,
+    )
+
+    num_ROIs = len(list_indices_ref)
+    compute = True  # TODO: Check whether compute is needed
```

Comment:
```python
compute = True # TODO: Check whether compute is needed
```
My first guess is that it's indeed needed.

https://github.com/scikit-image/scikit-image/blob/v0.21.0/skimage/registration/_phase_cross_correlation.py#L182-L418 specifies its input types as "array", but within the function it uses plenty of numpy operations. And for sure it returns numpy ndarrays, meaning that it is not lazy.
Thus I guess that even if it accepts dask arrays (which I'm not sure about) it would transform them into numpy at some point before returning.

Reply:
Yes indeed, turned out compute=True was the way to go here